src/os/filestore/FileStore.cc

```diff
@@ -2997,6 +2997,10 @@ void FileStore::_do_transaction(
 	  msg = "ENOTEMPTY suggests garbage data in osd data dir";
 	}
 
+        if (r == -EPERM) {
```

Comment:
personally, i'd go with an `else if` branching off after the previous if. otherwise it looks good to me.